src/Icon/Icon.tsx

```diff
@@ -0,0 +1,91 @@
+import { ComponentProps, FC, forwardRef, LegacyRef } from 'react';
+import { ClassNames, css, Theme, withTheme } from '@emotion/react';
+import styled from '@emotion/styled';
+import {
+  color,
+  ColorProps,
+  display,
+  DisplayProps,
+  fontSize,
+  FontSizeProps,
+  opacity,
+  OpacityProps,
+  space,
+  SpaceProps,
+  textAlign,
+  TextAlignProps,
+  textShadow,
+  TextShadowProps
+} from 'styled-system';
+
+type Rotation = '90' | '180' | '270' | null;
+
+const rotationStyles = ({ rotation }: { rotation?: Rotation }) =>
+  rotation &&
+  css`
+    transform: rotate(${rotation}deg);
+  `;
+
+type RotationProps = {
+  rotation?: Rotation;
+};
+
+const StyledIcon = styled.i<
+  ColorProps &
+    DisplayProps &
+    FontSizeProps &
+    SpaceProps &
+    TextAlignProps &
+    RotationProps &
+    TextShadowProps &
+    OpacityProps
+>`
+  ${color};
+  ${display};
+  ${fontSize};
+  ${space};
+  ${textAlign};
+  ${rotationStyles};
+  ${textShadow};
+  ${opacity};
+`;
+
+type Props = ComponentProps<typeof StyledIcon> & {
+  className?: string;
+  iconRef?: LegacyRef<HTMLElement>;
+  name: string;
+  rotation?: Rotation;
+  theme: Theme;
+};
+
+const Icon: FC<Props> = ({
+  className,
+  iconRef,
+  name,
+  rotation = null,
```

Comment:
Keeping this `null` default for now to match previous `defaultProps` as closely as possible.  We can revisit this later.